client/src/js/components/bhReportSource.js

```diff
@@ -0,0 +1,47 @@
+angular.module('bhima.components')
+.component('bhReportSource', {
+  bindings : {
+    onSelectCallback : '&?',
```

Comment:
I don't think this should be marked as optional - the component will likely error if you don't pass it in.